Assets/Scripts/EnemyServices/EnemyStates/EnemyAttackingState.cs

```diff
@@ -0,0 +1,89 @@
+﻿using UnityEngine;
+using BulletServices;
+
+namespace EnemyTankServices
+{
+    public class EnemyAttackingState : EnemyStates
+    {
+        public override void OnStateEnter()
+        {
+            base.OnStateEnter();
+            tankView.activeState = EnemyState.Attacking; 
+        }
+
+        private void Update()
+        {
+            if (!tankModel.b_PlayerInSightRange && !tankModel.b_PlayerInAttackRange) tankView.currentState.ChangeState(tankView.patrollingState);
+            else if (tankModel.b_PlayerInSightRange && !tankModel.b_PlayerInAttackRange) tankView.currentState.ChangeState(tankView.chasingState);
+
+            AttackPlayer();
```

Comment:
AttackPlayer(); we are doing this code in update it needs to be restricted  as update runs once per frame .